### PR TITLE
Test windows on py3.10rc1 and pypy3.7

### PR DESCRIPTION
**Pull request by graingert**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

### CODE REVIEW

**Comment by graingert** [comment]
I think this would be much easier with poetry https://github.com/celery/celery/pull/6680

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -26,13 +26,6 @@ jobs:
       matrix:
         python-version: ['3.6', '3.7', '3.8', '3.9', '3.10.0-beta.4', 'pypy-3.6', 'pypy-3.7']
         os: ["ubuntu-20.04", "windows-2019"]
```
Comment (graingert) [inline .github/workflows/python-package.yml:28]:
```suggestion
        os: ["ubuntu-20.04", "windows-2019"]
        exclude:
          - os: windows-2019
            python-version: 'pypy-3.7'
          - os: windows-2019
            python-version: 'pypy-3.6'
```

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -65,7 +58,10 @@ jobs:
         Run tox for
         "${{ matrix.python-version }}-unit"
       timeout-minutes: 15
-      run: >
+      env:
+        CRYPTOGRAPHY_DONT_BUILD_RUST: ${{ startsWith(matrix.python-version, 'pypy') && startsWith(matrix.os, 'windows-') && 1 }}
+      run: |
+        echo $CRYPTOGRAPHY_DONT_BUILD_RUST
```
Comment (graingert) [inline .github/workflows/python-package.yml:64]:
```suggestion
      run: |
```

File: tox.ini [hunk 3]
```diff
@@ -25,6 +25,7 @@ sitepackages = False
 recreate = False
 passenv =
     AZUREBLOCKBLOB_URL
+    CRYPTOGRAPHY_DONT_BUILD_RUST
```
Comment (graingert) [inline tox.ini:28]:
```suggestion
```

File: t/unit/backends/test_asynchronous.py [hunk 4]
```diff
@@ -141,6 +142,10 @@ def test_drain_timeout(self):
         assert on_interval.call_count < 20, 'Should have limited number of calls to on_interval'
 
 
+@pytest.mark.skipif(
+    sys.version_info >= (3, 9) and sys.platform == "win32",
```
Comment (graingert) [inline t/unit/backends/test_asynchronous.py:146]:
```suggestion
    sys.platform == "win32",
```

File: t/unit/backends/test_asynchronous.py [hunk 5]
```diff
@@ -141,6 +142,10 @@ def test_drain_timeout(self):
...
+    reason="hangs forever intermittentlhy on windows",
```
Comment (graingert) [inline t/unit/backends/test_asynchronous.py:147]:
```suggestion
    reason="hangs forever intermittently on windows"
```

File: .github/workflows/python-package.yml [hunk 6]
```diff
@@ -27,12 +27,8 @@ jobs:
         python-version: ['3.6', '3.7', '3.8', '3.9', '3.10.0-beta.4', 'pypy-3.6', 'pypy-3.7']
```
Comment (graingert) [inline .github/workflows/python-package.yml:27]:
```suggestion
        python-version: ['3.6', '3.7', '3.8', '3.9', '3.10.0-rc.1', 'pypy-3.6', 'pypy-3.7']
```

File: .github/workflows/python-package.yml [hunk 7]
```diff
@@ -27,12 +27,8 @@ jobs:
...
         os: ["ubuntu-20.04", "windows-2019"]
         exclude:
-          - os: windows-2019
-            python-version: 'pypy-3.7'
           - os: windows-2019
             python-version: 'pypy-3.6'
-          - os: windows-2019
-            python-version: "3.10.0-beta.4"
     continue-on-error: ${{ startsWith(matrix.python-version, '3.10.0-beta.') }}
```
Comment (graingert) [inline .github/workflows/python-package.yml:32]:
```suggestion
```

**Comment by graingert** [comment]
@auvipy this one is ready now

**Review by auvipy — APPROVED** [review]

**Pull request merged**